scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala

```diff
@@ -79,14 +79,25 @@ private[types] object TypeProvider {
     checkMacroEnclosed(c)
 
     val (r, caseClassTree, name) = annottees.map(_.tree) match {
-      case List(q"case class $name(..$fields) { ..$body }") =>
+      case l @ List(q"$mods class $name[..$tparams] $ctorMods(..$fields) extends { ..$earlydefns } with ..$parents { $self => ..$body }") if mods.asInstanceOf[Modifiers].hasFlag(Flag.CASE) =>
```

Comment:
Are all of these necessary? It's just the `@description` annotation you need right? Do we want to allow other traits on `toTable` classes?